docs/index.rst

```diff
@@ -106,24 +106,81 @@ You can use pip and make to quickly set up a development environment::
-* Document your code. Documentation can be easily generated by executing ``make
-  docs-html``.
-* Unit tests are *very* highly recommended.
+* All PR’s should follow the predetermined pull request template, explaining the issue, same follows for the Issue if one is to be filed.
+* Recommending squash and merge and rebase and merge and disable "standard merge" on the repository.
+* Maintain Coding Standards
+    * Keep pep8 rules
+    * Follow the same stylistic and logical patterns used in the code
+        * All entity class names and class attributes have to be in the singular format
+        * All required entity attributes have to have `required=True` parameter
+        * It is preferable to use `alpha` data type for default string values for easier debug procedure
+        * In case any workaround is introduced, it is necessary to provide corresponding BZ ID directly into the code docstring
+        * All linting and formatting/style checks would be enforced by Travis-CI and PR would be considered broken until checks are passed successfully.
 * Adhere to typical commit guidelines:
-
-    * Commits should not cause NailGun to break.
+    * Commits should not cause NailGun’s unit test to fail. If it does, it will the responsibility of contributor to fix them.
     * Commits should be small and coherent. One commit should address one issue.
-    * Commits should have `good commit messages`_.
-    * `Rebasing`_ is encouraged. Rebasing produces a much nicer commit history
-      than merging.
-
+    * Commits should have good commit messages.
+    * Rebasing is encouraged. Rebasing produces a much nicer commit history than merging.
+* The output of entity operations is a must, either use Satellite or (Foreman+Katello). Outputs should be provided for every new/affected helper or field in the entity. Or, if you have some tests exercising corresponding fields/helpers (for robottelo or other repo) - providing just test results is enough. This should make the review process easier for all reviewers and anyone else interested in the new functionality.
+* Follow the predetermined template to raise a PR or Issue. This can be used by reviewers while processing code changes.
+* When raising a PR against any other branch which is also applicable for '6.X' or foreman branch. Please specify that information in the PR message
+* Unit tests
+    * Unit tests are compulsory
+    * Unit tests should cover all available actions, for the entity. For eg: Repository Sets, have enable, disable, list_available there should be unit tests exercising these actions.
 * When in doubt, ask on IRC.
 
+**Important to Note :**
+
+* Define Foreman Version labels in Nailgun
+    The moment PR would be raised, contributor needs to set the right version.
+
+* All PRs should be raised along with Unit tests
+    The unit tests should be added  while adding a new entity or modifying the existing entity or modifying and adding to the core of Nailgun.
```

Comment:
I see that this hasn't been changed in newest commit. Can we have discussion on that before merge? @san7ket  @evgeni @sean797

Reply:
@mirzal Sorry I missed to send you the discussion document, Sent it to you. I think we had already came to conclusion that, It is good to have this as with increasing contributions would increase the risk of breakages and as of today we just rely on unit tests to check the sanity of nailgun

Reply:
Thanks for sharing! I didn't follow discussions around that document.

Let's put it this way: I disagree with the wording that we use right now, but re-reading it, I can see how it can be understood a bit differently. Since we had opportunity to discuss that already back in November, I take it that majority of people are OK with current state. And as I agree with principle/spirit of that rule, I don't see a reason to block this PR any further.

Let's go ahead with what we have. If specific wording becomes a problem in the future, we can address it then.